### PR TITLE
Added dotnet core 3.0 support

### PR DESCRIPTION
**Pull request by Lakritzator**
For this I had to:
* disable paket (doesn't support netcoreapp3.0) and use package references
* add the target netcoreapp3.0 and use FrameworkReference Microsoft.DesktopUI for this
* change the SDK from MSBuild.Sdk.Extras to Microsoft.NET.Sdk.Wpf, I also had to add *.xaml mapping (Directory.build.props) -> check what side effects this has, as I didn't use MSBuild.Sdk.Extras before.
* remove the sdk from global.json (so it takes the latest)
* Added installing the SDK to appveyor.yml

I was able to run all targets (net45, net462 and netcoreapp3.0) and couldn't see any visible changes.
The question is, do you need paket or wouldn't you rather change to the build in package reference?
Anyway, this is up to you!

### CODE REVIEW

**Comment by Lakritzator** [comment]
Hmm... was expected that if I need to disable paket for local builds, it also needs to be disabled for appveyor builds!

**Comment by punker76** [comment]
@Lakritzator removing the paket.reference files should help here

**Comment by punker76** [comment]
@Lakritzator or better commenting the content, so we can leave the files for later (if paket works together with netcore30)

File: src/ControlzEx.Showcase/ControlzEx.Showcase.csproj [hunk 1]
```diff
@@ -1,26 +1,38 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project Sdk="MSBuild.Sdk.Extras">
+<Project Sdk="Microsoft.NET.Sdk.Wpf">
     <!-- Project properties -->
     <PropertyGroup>
+        <TargetFrameworks>net45;net462;netcoreapp3.0</TargetFrameworks>
```
Comment (batzen) [inline src/ControlzEx.Showcase/ControlzEx.Showcase.csproj]: Why did you move this to the projects instead of relying on the value from Directory.Build.props?

Comment (Lakritzator) [reply src/ControlzEx.Showcase/ControlzEx.Showcase.csproj]: Good question: I noticed on my projects that I didn't get the build working, if the TargetFrameworks are specified in the Directory.Build.props. The same happened here, you get many weird errors!
I will actually address this to Microsoft in my next status email, maybe they can have a look it it.

P.S.
I now assume it has to do with the SDK "Microsoft.NET.Sdk.Wpf" vs "MSBuild.Sdk.Extras"...

File: src/ControlzEx.Showcase/ControlzEx.Showcase.csproj [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
     <ItemGroup>
         <Reference Include="System.Windows.Interactivity">
             <HintPath>..\ExtLibs\NET45\System.Windows.Interactivity.dll</HintPath>
         </Reference>
-        <None Include="app.manifest"/>
     </ItemGroup>
-    <Import Project="..\..\.paket\Paket.Restore.targets"/>
+    <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
+        <FrameworkReference Include="Microsoft.DesktopUI" />
+        <PackageReference Include="System.Management" Version="4.5.0" />
```
[the remark this answers is not in the document]
Comment (Lakritzator) [reply src/ControlzEx.Showcase/ControlzEx.Showcase.csproj]: You seem to be right, but it is imported in src\ControlzEx\Behaviors\WindowChromeBehavior.cs
The import is not used though...

**Comment by Lakritzator** [comment]
> @Lakritzator removing the paket.reference files should help here

Isn't the problem in the build.cake file? I guess that PaketPack will also fail?
As I said before, I have 0 experience with Paket, NuGet used to do what I needed...

**Comment by punker76** [comment]
@Lakritzator ok, then comment the pack task in the cake script and I will change the packaging to msbuild packaging. All other should work in the cake script (hopefully).

File: src/ControlzEx.Showcase/ControlzEx.Showcase.csproj (hunk 1 above)
Comment (batzen) [reply src/ControlzEx.Showcase/ControlzEx.Showcase.csproj]: You might be right with your guess as it wasn't working in older versions of the Extras-SDK and was fixed some time ago. Maybe it's the same issue with the WPF-SDK.

**Comment by batzen** [comment]
We used paket because it has lock files, doesn't require to mention the desired version on each reference but in one central place instead, takes care of framework specific references, allows grouping of dependencies and defining which package sources should be used is (was?) a lot easier than with nuget itself.

To use the msbuild packaging we would also need something like gitversion there, don't we?

**Comment by punker76** [comment]
>To use the msbuild packaging we would also need something like gitversion there, don't we?

@batzen I do this already in MahApps and IconPacks like you in Fluent ;-)

**Comment by Lakritzator** [comment]
I'll see if I can make it work without paket, for gitversion there is something too...

**Comment by punker76** [comment]
@Lakritzator Put these in the csproj

```
<!-- NuGet -->
<PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
    <Authors>Jan Karger, Bastian Schmidt, James Willock</Authors>
    <PackageLicenseUrl>https://github.com/ControlzEx/ControlzEx/blob/develop/LICENSE</PackageLicenseUrl>
    <PackageProjectUrl>https://github.com/ControlzEx/ControlzEx</PackageProjectUrl>
    <PackageIconUrl>https://raw.githubusercontent.com/ControlzEx/ControlzEx/develop/logo-mini.png</PackageIconUrl>
    <PackageTags>WPF;UI;Metro;Modern;Material;XAML;Toolkit;Library;.NET;OSS;OpenSource;WindowChrome</PackageTags>
    <PackageReleaseNotes>https://github.com/ControlzEx/ControlzEx/releases</PackageReleaseNotes>
    <RepositoryUrl>https://github.com/ControlzEx/ControlzEx.git</RepositoryUrl>
    <RepositoryType>git</RepositoryType>
    <GenerateLibraryLayout>true</GenerateLibraryLayout>
</PropertyGroup>
```

and this task in the cake script

```
Task("Pack")
  .WithCriteria(() => !isPullRequest)
    .Does(() =>
{
    var msBuildSettings = new MSBuildSettings { ToolPath = msBuildPath };
    var project = "./src/ControlzEx/ControlzEx.csproj";

    MSBuild(project, msBuildSettings
      .SetConfiguration(configuration)
      .SetVerbosity(Verbosity.Normal)
      .WithTarget("pack")
      .WithProperty("PackageOutputPath", "bin")
      .WithProperty("RepositoryBranch", branchName)
      .WithProperty("RepositoryCommit", gitVersion.Sha)
      .WithProperty("Description", "ControlzEx is a library with some shared Controls for WPF.")
      .WithProperty("Version", isReleaseBranch ? gitVersion.MajorMinorPatch : gitVersion.NuGetVersion)
      .WithProperty("AssemblyVersion", gitVersion.AssemblySemVer)
      .WithProperty("FileVersion", gitVersion.AssemblySemFileVer)
      .WithProperty("InformationalVersion", gitVersion.InformationalVersion)
    );
});
```

**Comment by Lakritzator** [comment]
So, another try...

This commit was locally tested with .\build.ps1 -Target appveyor, a nuget package was produced!!

I _slightly_ modified the cake script, hope that is okay with you.
Had to fix the build.ps1 too, as this used paket, using version from the internet which I modified to work from behind a coperate firewall.

**Comment by Lakritzator** [comment]
Build worked...

As there is no package build for PR's, don't know why, I checked locally.

Nuspec:
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>ControlzEx</id>
    <version>4.0.0-dotnetcore30-0200</version>
    <authors>Jan Karger, Bastian Schmidt, James Willock</authors>
    <owners>Jan Karger, Bastian Schmidt, James Willock</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <licenseUrl>https://github.com/ControlzEx/ControlzEx/blob/develop/LICENSE</licenseUrl>
    <projectUrl>https://github.com/ControlzEx/ControlzEx</projectUrl>
    <iconUrl>https://raw.githubusercontent.com/ControlzEx/ControlzEx/develop/logo-mini.png</iconUrl>
    <description>ControlzEx is a library with some shared Controls for WPF.</description>
    <releaseNotes>https://github.com/ControlzEx/ControlzEx/releases</releaseNotes>
    <copyright>Copyright © 2015 - 2018 Jan Karger, Bastian Schmidt, James Willock</copyright>
    <tags>WPF UI Metro Modern Material XAML Toolkit Library .NET OSS OpenSource WindowChrome</tags>
    <repository type="git" url="https://github.com/ControlzEx/ControlzEx.git" branch="feature/dotnetcore30" commit="1485023febdbd511e9acdd7f9c50a9ab6b7ddc21" />
    <dependencies>
      <group targetFramework=".NETFramework4.5">
        <dependency id="JetBrains.Annotations" version="2018.2.1" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETFramework4.6.1">
        <dependency id="JetBrains.Annotations" version="2018.2.1" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETCoreApp3.0">
        <dependency id="JetBrains.Annotations" version="2018.2.1" exclude="Build,Analyzers" />
      </group>
    </dependencies>
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.CSharp" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="PresentationCore" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="PresentationFramework" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Data.DataSetExtensions" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Data" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Net.Http" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Windows.Forms" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Xaml" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Xml" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Xml.Linq" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="WindowsBase" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
    </frameworkAssemblies>
  </metadata>
</package>
```
I actually see a few things which might need some optimizing:

1. JetBrains.Annotations is still in the nuspec
2. There are frameworkAssembly references which are probably not needed...

Anything else?

**Comment by punker76** [comment]
@Lakritzator The PackageReference needs `PrivateAssets="All"`

**Comment by Lakritzator** [comment]
Yes I know :) Just asking what else you need changeing.

**Comment by punker76** [comment]
@Lakritzator The appveyor nuget feed gets spamed if we build a nuget package on every pr build

**Comment by Lakritzator** [comment]
Fail enough, I guess this has never been an issue for me as no-one wants to PR for my projects :grin:

This is how it looks now, satisfied?

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
  <metadata>
    <id>ControlzEx</id>
    <version>4.0.0-dotnetcore30-0201</version>
    <authors>Jan Karger, Bastian Schmidt, James Willock</authors>
    <owners>Jan Karger, Bastian Schmidt, James Willock</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <licenseUrl>https://github.com/ControlzEx/ControlzEx/blob/develop/LICENSE</licenseUrl>
    <projectUrl>https://github.com/ControlzEx/ControlzEx</projectUrl>
    <iconUrl>https://raw.githubusercontent.com/ControlzEx/ControlzEx/develop/logo-mini.png</iconUrl>
    <description>ControlzEx is a library with some shared Controls for WPF.</description>
    <releaseNotes>https://github.com/ControlzEx/ControlzEx/releases</releaseNotes>
    <copyright>Copyright © 2015 - 2018 Jan Karger, Bastian Schmidt, James Willock</copyright>
    <tags>WPF UI Metro Modern Material XAML Toolkit Library .NET OSS OpenSource WindowChrome</tags>
    <repository type="git" url="https://github.com/ControlzEx/ControlzEx.git" branch="feature/dotnetcore30" commit="5fe0f8b8de24b4a3f40f296cebcfbde7c463a205" />
    <dependencies>
      <group targetFramework=".NETFramework4.5" />
      <group targetFramework=".NETFramework4.6.1" />
      <group targetFramework=".NETCoreApp3.0" />
    </dependencies>
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.CSharp" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="PresentationCore" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="PresentationFramework" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Xaml" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="WindowsBase" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
    </frameworkAssemblies>
  </metadata>
</package>
```

**Comment by punker76** [comment]
@Lakritzator Love it!

**Comment by Lakritzator** [comment]
I think some more optimizing of the .props and .csproj is possible, don't know how much time we should invest?

I also don't know if the application really does what it should... that is for you to look at...

Do you want me to compackt my commits, to make the PR into one?

**Comment by Lakritzator** [comment]
One important remark, the build will take a bit longer as dotnet core 3.0 needs to be installed!
As soon as dotnet core 3.0 is made publicly available I assume AppVeyor has this pre-installed.

**Comment by Lakritzator** [comment]
Also, I (or we) might need to contact Oren Novotny for support in MSBuild.Sdk.Extras!

**Comment by punker76** [comment]
@batzen This one LGTM

**Comment by punker76** [comment]
@Lakritzator 

>I also don't know if the application really does what it should... that is for you to look at...

It runs...

>Do you want me to compackt my commits, to make the PR into one?

I think this is not necessary

**Comment by Lakritzator** [comment]
FYI: If you are updating Visual Studio 2017 to 15.9.x, don't forget to read this: https://blogs.msdn.microsoft.com/dotnet/2018/11/13/net-core-tooling-update-for-visual-studio-2017-version-15-9/
Otherwise the dotnet core 3.0 SDK doesn't work!!

**Comment by batzen** [comment]
> @Lakritzator The appveyor nuget feed gets spamed if we build a nuget package on every pr build

That's not true, as we already prevent publishing on PRs in the appveyor config through
```
  nuget:
    disable_publish_on_pr: true
```

> Also, I (or we) might need to contact Oren Novotny for support in MSBuild.Sdk.Extras!

I don't think we need the extras-sdk once microsoft has finished adding support for wpf.


> I'll see if I can make it work without paket, for gitversion there is something too...

I started using EmbedAllSources in Fluent.Ribbon instead of relying on gitlink.
Using EmbedAllSources will add the source code directly to the PDBs instead of pointing to github.

File: build.cake [hunk 3]
```diff
@@ -99,26 +94,32 @@ Task("Clean")
...
-
-    var msBuildSettings = new MSBuildSettings { ToolPath = msBuildPath, ArgumentCustomization = args => args.Append("/m") };
-    MSBuild(solution, msBuildSettings
-            .SetConfiguration(configuration)
-            .SetVerbosity(Verbosity.Minimal)
-            .WithTarget("restore")
-            );
+    var msBuildSettings = new MSBuildSettings {
+        Verbosity = Verbosity.Minimal,
+        ToolVersion = MSBuildToolVersion.VS2017,
```
Comment (batzen) [inline build.cake]: Why do we have to specify this explicitly?

File: build.cake [hunk 4]
```diff
@@ -127,13 +128,28 @@ Task("Build")
 });
 
 Task("Pack")
+    .WithCriteria(() => !isPullRequest)
```
Comment (batzen) [inline build.cake]: Pack should also work on PRs. Publishing of resulting artifacts is prevent by appveyor config.

File: src/Directory.build.props [hunk 5]
```diff
@@ -1,20 +1,16 @@
...
     <!-- Project properties -->
     <PropertyGroup>
-        <TargetFrameworks>net45;net462</TargetFrameworks>
-        <ExtrasEnableWpfProjectSetup>true</ExtrasEnableWpfProjectSetup>
-        <ExtrasDefaultResourceIncludes>Resources\*.xaml</ExtrasDefaultResourceIncludes>
         <OutputPath>$(MSBuildThisFileDirectory)\bin\$(Configuration)\$(MSBuildProjectName)</OutputPath>
         <AppendTargetFrameworkToOutputPath>true</AppendTargetFrameworkToOutputPath>
         <GenerateDocumentationFile>false</GenerateDocumentationFile>
         <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+		<DisableImplicitFrameworkReferences Condition="'$(TargetFramework)' != 'netcoreapp3.0'">true</DisableImplicitFrameworkReferences>
```
Comment (batzen) [inline src/Directory.build.props]: Why do we need this?

File: build.cake (hunk 3 above)
Comment (Lakritzator) [reply build.cake]: If you mean the tool version, this was added so we don't need the additional tool dependency on "VSwhere". If you mean the MSBuildSettings, they were already there and I find this way more easy to read as the fluent calls which where there. It separates the how and what nicely.

File: build.cake (hunk 4 above)
Comment (Lakritzator) [reply build.cake]: There is a difference between publishing to nuget and placing it on the AppVeyor feed, the first is prevented the second is not.

I agree with Jan, if you have a lot of PR's you don't want artifacts in the project feed to appear on every PR. If there is a setting which prevents the artifact from being placed on the feed, I didn't look into this, than I would also remove the criteria. The package would than be available on the artifacts page of the job.

File: src/Directory.build.props (hunk 5 above)
Comment (Lakritzator) [reply src/Directory.build.props]: This removes the standard Framework references, like System.Xml, System.Xml.Linq and gives you control over what Framework references are used. It reduces the clutter in the project, maybe even reduce build time, and makes the nupkg a bit cleaner.

On the other side, you need to specify the used FrameworkReferences manually.

It depends a bit on how much control one wants... doesn't really hurt to remove it, than you can remove the Reference entries for ! netcoreapp3.0 further ahead.

File: build.cake (hunk 3 above)
Comment (batzen) [reply build.cake]: I was talking about the toolversion.
You can safely remove the explicit toolversion as the previous usage of vswhere was mistakenly added by @punker76 as far as i remember.

Regarding the other settings, why do we need to state the PlatformTarget here?

File: build.cake (hunk 4 above)
Comment (batzen) [reply build.cake]: > There is a difference between publishing to nuget and placing it on the AppVeyor feed, the first is prevented the second is not.

There is no auto publish for nuget here, so it prevents the second.
So WithCriteria here is not needed.

Documentation from appveyor:
>   disable_publish_on_pr: true     # disable publishing of .nupkg artifacts to
                                  # account/project feeds for pull request builds

File: src/Directory.build.props [hunk 6]
```diff
@@ -34,4 +30,26 @@
         <!-- https://github.com/dotnet/sourcelink/blob/master/docs/README.md#embedallsources -->
         <EmbedAllSources>true</EmbedAllSources>
     </PropertyGroup>
+    <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
```
Comment (batzen) [inline src/Directory.build.props]: Aren't the blocks below exactly the same for each framework?

File: src/ControlzEx/ControlzEx.csproj [hunk 7]
```diff
@@ -16,9 +30,12 @@
     </PropertyGroup>
     <!-- Items include -->
     <ItemGroup>
+        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All"/>
         <Reference Include="System.Windows.Interactivity">
             <HintPath>..\ExtLibs\NET45\System.Windows.Interactivity.dll</HintPath>
         </Reference>
     </ItemGroup>
-    <Import Project="..\..\.paket\Paket.Restore.targets"/>
+    <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
```
Comment (batzen) [inline src/ControlzEx/ControlzEx.csproj]: Shouldn't this block be in Direcoty.build.props?
The full framework references are placed there also. ;-)

File: src/ControlzEx/ControlzEx.csproj [hunk 8]
```diff
@@ -16,9 +30,12 @@
     </PropertyGroup>
     <!-- Items include -->
     <ItemGroup>
+        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All"/>
```
Comment (batzen) [inline src/ControlzEx/ControlzEx.csproj]: Do we need this if we use EmbedAllSources?

Comment (Lakritzator) [reply src/ControlzEx/ControlzEx.csproj]: That I cannot answer, that was suggested by @punker76

File: src/ControlzEx/ControlzEx.csproj (hunk 7 above)
Comment (Lakritzator) [reply src/ControlzEx/ControlzEx.csproj]: Yeah, moving it

File: src/Directory.build.props (hunk 6 above)
Comment (Lakritzator) [reply src/Directory.build.props]: Yes, reducing them... in an initial (pre-commit) state there was something else.

File: build.cake (hunk 4 above)
Comment (Lakritzator) [reply build.cake]: You convinced me, I removed the criteria.

**Comment by Lakritzator** [comment]
Hmmm:

Successfully created package 'C:\projects\controlzex\src\ControlzEx\bin\ControlzEx.4.0.0-pullrequest0066-0204.nupkg'.

And later:
No artifacts found matching 'src\bin\ControlzEx*.nupkg' path

File: build.cake [hunk 9]
```diff
@@ -129,11 +130,25 @@ Task("Build")
...
+        Verbosity = Verbosity.Normal,
+        ToolVersion = MSBuildToolVersion.VS2017,
+        Configuration = configuration,
+        PlatformTarget = PlatformTarget.MSIL
+    };
+    var project = "./src/ControlzEx/ControlzEx.csproj";
+
+    MSBuild(project, msBuildSettings
+      .WithTarget("pack")
+      .WithProperty("PackageOutputPath", "bin")
```
Comment (batzen) [inline build.cake]: This previously used `var packDestDir = "src/bin"`, now it seems to use `bin` relative to the project being built.
That should also explain the missing nupkg on appveyor.

File: src/ControlzEx/ControlzEx.csproj (hunk 8 above)
Comment (punker76) [reply src/ControlzEx/ControlzEx.csproj]: If we now use `<EmbedAllSources>true</EmbedAllSources>` then we can remove the `SourceLink` reference.

File: build.cake (hunk 9 above)
Comment (Lakritzator) [reply build.cake]: Fixed path, should now be okay

File: src/ControlzEx/ControlzEx.csproj (hunk 8 above)
Comment (Lakritzator) [reply src/ControlzEx/ControlzEx.csproj]: I always thought there is a huge difference between placing all the sources in the dll, vs. linking the symbols to a github repository... Is that really what you want?

File: src/ControlzEx.Showcase/ControlzEx.Showcase.csproj (hunk 1 above)
Comment (Lakritzator) [reply src/ControlzEx.Showcase/ControlzEx.Showcase.csproj]: I think this is not something we can fix now, closing the conversation...

File: build.cake (hunk 3 above)
Comment (Lakritzator) [reply build.cake]: I'm not sure if it's the default, but we specify the configuration and the platform target together.
Maybe it's just because I like to be "verbose" and show what I want...

You decide what you want, without ToolVersion and PlatformTarget?
Not sure it works...

**Comment by Lakritzator** [comment]
The nuget package is now created on the correct location,

File: src/ControlzEx/ControlzEx.csproj (hunk 8 above)
Comment (punker76) [reply src/ControlzEx/ControlzEx.csproj]: I'm not sure. `EmbedAllSources` is new to me and until yet I used gitlink or SourceLink. EmbedAllSources has the benefit of faster debugging, but this is not importend for me, I can live with gitlink or SourceLink.

Comment (Lakritzator) [reply src/ControlzEx/ControlzEx.csproj]: Your decision, but I would like to be able to continue my dotnet core 3.0 work with other repositories... so I hope we can close this PR soon :question:

File: build.cake (hunk 3 above)
Comment (batzen) [reply build.cake]: Being explicit here would require a change here when we want to use VS2019 ;-)

**Comment by batzen** [comment]
@punker76 LGTM

**Comment by punker76** [comment]
@Lakritzator Thank you very much 👍 😃

**Comment by Lakritzator** [comment]
You are more than welcome, it's the least I can do if I will be using your software.
